datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -411,7 +411,7 @@ def log_typos_in_options(self, user_config, models_config, level):
         models_config = models_config or {}
         typos = set()  # type: Set[str]
 
-        known_options = dict(models_config).keys()
+        known_options = [k for k, _ in models_config]  # type List[str]
         unknown_options = sorted(list(user_config.keys() - known_options))
```

Comment:
Could this issue potentially happen if a user provided a wrong config option called `keys`?

Reply:
No, it won't because `user_config` is already a dictionary and isn't being converted using `dict()`